### DISCUSSION TITLE
start/stop code  and frequency input

### DISCUSSION THREAD

**Discussion by holgerlembke** [dbuezas/arduino-web-timers#36] [Ideas]
for Arduino ATMEGA328 I use http://www.8bit-era.cz/arduino-timer-interrupts-calculator.html when I need a timer with a specific frequency. I like it, because it is a down to the point solution: input cpu freq and desired frequency, bingo, code, done.

AWT needs more thinking about what I want and what I need to click... Might be overwhelming for the casual user.

My goal was to get a timer3 with 1000 Hz on a 16 MHz lgt8f328p.

-- First problem: any way to enter 1000 Hz? I found none, I think AWT works the other way round only.

-- I wanted start/stop the timer, so I need start and stop code.

```
void startTimer(void) {
  noInterrupts();
  TCCR3A = 1 << COM3A1;
  TCCR3B = 1 << WGM32 | 1 << CS30;
  TIMSK3 = 1 << OCIE3A;  // set Timer3 Interrupt
  TIFR3 = 1 << OCF3A;     //<<<<<!!!!!! needs to clear interrupt
  DDRF = 1 << DDF1;
  OCR3A = 16000;  // 1 kHz with 16 mhz?
  interrupts();
}

void stopTimer() {
  cli();
  TIMSK3 = 0;  // Disable interrupt
  sei();
}
```

**Comment by dbuezas** [comment]
Hopefully this tool helps getting a deeper understanding of how timers work, all they can do, and the limitations of each mode.
If you ever wondered why there's more than one mode, or why your pwm isn't going all the way down to 0% or all the way up to 100%, or what extra hidden features there are, then this tool can be useful. If you just need an interrupt every ms then it is a bit of an overkill.
